### PR TITLE
tiltfile: local: add stdin param

### PR DESCRIPTION
**Pull request by landism**
This is mostly to make it easier to get a starlark variable into a command (e.g., `kubectl apply -f -` or `cat > foo.yaml`).

This is currently possible by escaping the var into the command line and piping, but 1) escaping feels finicky and 2) that risks hitting ARG_MAX.

### CODE REVIEW

File: internal/tiltfile/api/__init__.py [hunk 1]
```diff
@@ -675,6 +676,7 @@ def local(command: Union[str, List[str]],
     echo_off: If set to True, skips printing command to log.
     env: Environment variables to pass to the executed ``command``. Values specified here will override any variables passed to the Tilt parent process.
     dir: Working directory for ``command``. Defaults to the Tiltfile's location.
+    stdin: If non-empty, will be written to ``command``'s stdin.
```
Comment (landism) [inline internal/tiltfile/api/__init__.py:679]: Or maybe #3442 will want an empty stdin so this should be "if non-None"

Comment (nicks) [reply internal/tiltfile/api/__init__.py:679]: ya, we should handle None and empty string differently? i think there are some subtle differences in how the underlying process APIs handle it but I'm not 100% sure.

**Review by milas — APPROVED** [review]
Agree it's probably advisable to distinguish `None` from `""` (or provide some sort of special built-in value like `DUMMY_STDIN`), but otherwise 👍

**Comment by landism** [comment]
redid it using `None` instead

and generics!

**Review by nicksieger — APPROVED** [review]
❤️

**Pull request merged**